### ISSUE TITLE
work associations panel optmization

### ISSUE THREAD

**Issue by dertel** [dertel/old-issues#509]
**[Original report](https://bitbucket.org/davidartrecord/artrecord/issues/509) by David Ertel (Bitbucket: [davidartrecord](https://bitbucket.org/davidartrecord), GitHub: [dertel](https://github.com/dertel)).**

----------------------------------------

```diff
diff --git a/app/models/concerns/common.rb b/app/models/concerns/common.rb
index 2cc3f35..686459e 100644
--- a/app/models/concerns/common.rb
+++ b/app/models/concerns/common.rb
@@ -89,14 +89,12 @@ module Common
     @get_images = [] if @get_images.blank?
 
     if self.try(:image_list)
-      @get_images[self.id] = self.image_list.split(",").map { |id|
-        Image.find_by(id: id.strip)
-      }.compact
+      @get_images[self.id] = Image.where(id: [self.image_list.split(",")])
     else
-      @get_images[self.id] = nil
+      @get_images[self.id.to_i] = nil
     end
 
-    @get_images[self.id]
+    @get_images[self.id.to_i]
   end
 
   #  used in application_helper#sanitize_snapshot
diff --git a/app/models/event.rb b/app/models/event.rb
index 0ce0a2f..70e698a 100644
--- a/app/models/event.rb
+++ b/app/models/event.rb
@@ -40,7 +40,11 @@ class Event < ApplicationRecord
   # sort by most recently added
   def works_by_sort_order
     if self.id.present?
-      Work.includes(:work_event_bonds).where(work_event_bonds: {event_id: self.id }).order("work_event_bonds.created_at")
+      Work.
+        includes(:work_event_bonds).
+        where(work_event_bonds: {event_id: self.id }).
+        order("work_event_bonds.created_at").
+        includes(:artists)
     end
   end
   
diff --git a/app/views/record/_association.html.erb b/app/views/record/_association.html.erb
index af3ac70..26aa5f8 100644
--- a/app/views/record/_association.html.erb
+++ b/app/views/record/_association.html.erb
@@ -24,6 +24,7 @@ else
 end
 %>
 
+<% cache [@record, @share, record_type, records.map{|r| [r.unique_id, r.updated_at]}] do %>
 <div class="<%= inline ? 'associations-meta ' : 'panel associations-panel ' %> <%= record_type %>-associations <%= panel_closed(record_type + '_assocs') %> <%= sortable %>">
   <% unless inline %>
   <div class="panel-title"><%= record_type_title %></div>
@@ -36,7 +37,8 @@ end
           <span class="association-remove-button">✕</span>
           <a href="<%= url_for(record) %>" data-id="<%= record.id %>">
             <% 
-              formatted_title = record.formatted_title(true, true).present? ? record.formatted_title(true, true) : record.unique_uid
+              formatted_title = record.formatted_title(true, true)
+              formatted_title = formatted_title.present? ? formatted_title : record.unique_uid
 
               if record.type == 'contact' and record.role.present?
                 formatted_title = "#{formatted_title} (#{record.role})"
@@ -57,3 +59,4 @@ end
   </div>
 
 </div>
+<% end #cache %>
```

**Comment by dertel** [comment]
**Original comment by David Ertel (Bitbucket: [davidartrecord](https://bitbucket.org/davidartrecord), GitHub: [dertel](https://github.com/dertel)).**

----------------------------------------

applied this to the homepage branch

**Comment by dertel** [comment]
**Original comment by David Ertel (Bitbucket: [davidartrecord](https://bitbucket.org/davidartrecord), GitHub: [dertel](https://github.com/dertel)).**

----------------------------------------

query optimizaion; more homepage tweaks, onboarding fix for gallery version, see #509
